### PR TITLE
feat(BuySell): add credit card limit to payment item

### PR DESCRIPTION
**Pull request by awiermanncasas-bc**
## Description
Update designs on the card, bank and wire transfer items in the payment methods and in settings page
Screen Shot 2022-07-12 at 2 22 43 PM
Screen Shot 2022-07-12 at 2 18 34 PM
Screen Shot 2022-07-12 at 9 58 01 AM
Screen Shot 2022-07-12 at 9 57 54 AM
Screen Shot 2022-07-12 at 9 57 47 AM

### CODE REVIEW

File: packages/blockchain-wallet-v4-frontend/src/scenes/Settings/General/LinkedBanks/template.success.tsx [hunk 1]
```diff
@@ -33,13 +31,30 @@ const StyledSettingsContainer = styled(SettingContainer)`
...
 const Success: React.FC<InjectedFormProps<{}, Props> & Props> = (props) => {
+  const { data: paymentMethods, isLoading: isLoadingPaymentMethods } = useRemote(
+    selectors.components.buySell.getBSPaymentMethods
+  )
+
+  const bankLimit = useMemo(() => {
+    if (isLoadingPaymentMethods) return
+
+    return (
+      paymentMethods?.methods.find((method) => method.type === BSPaymentTypes.BANK_TRANSFER)
```
Comment (mperdomo-bc) [inline packages/blockchain-wallet-v4-frontend/src/scenes/Settings/General/LinkedBanks/template.success.tsx:42]: Would it make more sense to assign this to a variable in the function and not returning it? At a glance this looks like a ternary, not that there is a default value being returned if no ´.limits´ is present or no `paymentMethods.methods` matching the find are present.
Also, why are these the default limits? I mean, this must be specified somewhere but it does look like magic numbers, it being "hidden" here.
Doing something like `return (bankPaymentMethod?.limits || DEFAULT_LIMIT)` can be a bit more expressive..

File: packages/blockchain-wallet-v4-frontend/src/scenes/Settings/General/LinkedBanks/template.success.tsx [hunk 2]
```diff
@@ -97,7 +169,7 @@ const Success: React.FC<InjectedFormProps<{}, Props> & Props> = (props) => {
                   >
                     <FormattedMessage id='buttons.remove' defaultMessage='Remove' />
                   </RemoveButton>
-                </Child>
+                </Child> */}
```
Comment (mperdomo-bc) [inline packages/blockchain-wallet-v4-frontend/src/scenes/Settings/General/LinkedBanks/template.success.tsx:172]: Is this commented code ok to be commented?

File: packages/blockchain-wallet-v4-frontend/src/components/Flyout/model.tsx [hunk 3]
```diff
@@ -582,6 +582,20 @@ const renderBankFullName = (
...
 ): string | ReactElement => {
+  if ('limits' in value) {
+    return (
+      <FormattedMessage
+        id='modals.simplebuy.bank_item_with_limits'
+        defaultMessage='{limitAmount} Limit'
+        values={{
+          limitAmount: fiatToString({
+            unit: value.currency,
+            value: convertBaseToStandard('FIAT', value.limits.max)
```
Comment (julianromero-bc) [inline packages/blockchain-wallet-v4-frontend/src/components/Flyout/model.tsx:593]: you could get FIAT string from Coin.FIAT so we prevent more hardcoded string

File: packages/blockchain-wallet-v4-frontend/src/modals/BuySell/LinkedPaymentAccounts/Accounts/Bank/index.tsx [hunk 4]
```diff
@@ -25,18 +29,47 @@ type Props = {
...
-    </MultiRowContainer>
-  </DisplayContainer>
-)
+const Bank = ({ icon, onClick, text, value }: Props) => {
+  const limitAmount = useMemo(() => {
+    if (!value.limits) return null
+
+    return fiatToString({
+      unit: value.currency,
+      value: convertBaseToStandard('FIAT', value.limits.max)
```
Comment (julianromero-bc) [inline packages/blockchain-wallet-v4-frontend/src/modals/BuySell/LinkedPaymentAccounts/Accounts/Bank/index.tsx:38]: And here

File: packages/blockchain-wallet-v4-frontend/src/modals/BuySell/LinkedPaymentAccounts/Accounts/Card/index.tsx [hunk 5]
```diff
@@ -23,27 +26,50 @@ type Props = {
...
-            defaultMessage='Credit or Debit Card'
-          />
-        )}
-      </StyledTitle>
-    </MultiRowContainer>
-  </DisplayContainer>
-)
+const Card: React.FC<Props> = ({ icon, onClick, text, value }) => {
+  const limitAmount = useMemo(() => {
+    if (!value.limits) return null
```
Comment (julianromero-bc) [inline packages/blockchain-wallet-v4-frontend/src/modals/BuySell/LinkedPaymentAccounts/Accounts/Card/index.tsx:31]: this value object needs destructuring

File: packages/blockchain-wallet-v4-frontend/src/scenes/Settings/General/LinkedBanks/template.success.tsx (hunk 1 above)
Comment (julianromero-bc) [reply packages/blockchain-wallet-v4-frontend/src/scenes/Settings/General/LinkedBanks/template.success.tsx:42]: Agree magic numbers should not be hidden in the code

File: packages/blockchain-wallet-v4-frontend/src/scenes/Settings/General/LinkedBanks/template.success.tsx (hunk 2 above)
Comment (julianromero-bc) [reply packages/blockchain-wallet-v4-frontend/src/scenes/Settings/General/LinkedBanks/template.success.tsx:172]: we should not commit commented code

File: packages/blockchain-wallet-v4-frontend/src/scenes/Settings/General/LinkedBanks/template.success.tsx (hunk 1 above)
Comment (awiermanncasas-bc) [reply packages/blockchain-wallet-v4-frontend/src/scenes/Settings/General/LinkedBanks/template.success.tsx:42]: actually, if the api is not providing any limits it should not show any limits at all, correct?

**Review by milan-bc — DISMISSED** [review]

File: packages/blockchain-wallet-v4-frontend/src/components/Flyout/model.tsx [hunk 6]
```diff
@@ -603,29 +618,56 @@ const renderBank = (value: BSPaymentMethodType | BankTransferAccountType | Benef
 )
 
 const renderCardText = (value: BSPaymentMethodType): string => {
-  return value.card
-    ? value.card.label
-      ? value.card.label.toLowerCase()
-      : value.card.type
-    : 'Credit or Debit Card'
+  if (value.card) {
```
Comment (julianromero-bc) [inline packages/blockchain-wallet-v4-frontend/src/components/Flyout/model.tsx:621]: getting card param several times, might be a good idea destructuring here

File: packages/blockchain-wallet-v4-frontend/src/modals/BuySell/LinkedPaymentAccounts/Accounts/Bank/index.tsx [hunk 7]
```diff
@@ -25,18 +30,51 @@ type Props = {
...
-  </DisplayContainer>
-)
+const Bank = ({ icon, onClick, text, value }: Props) => {
+  const limitAmount = useMemo(() => {
+    const { limits } = value
+
+    if (!limits) return null
+
+    return fiatToString({
+      unit: value.currency,
```
Comment (julianromero-bc) [inline packages/blockchain-wallet-v4-frontend/src/modals/BuySell/LinkedPaymentAccounts/Accounts/Bank/index.tsx:40]: why not currency in destructuring?

File: packages/blockchain-wallet-v4-frontend/src/modals/BuySell/LinkedPaymentAccounts/Accounts/Bank/index.tsx [hunk 8]
```diff
@@ -25,18 +30,51 @@ type Props = {
...
+      data-e2e={`sb${value.type.toLowerCase()}Banks`}
+      role='button'
+      onClick={onClick}
+    >
+      <DisplayIcon>{icon}</DisplayIcon>
+      <MultiRowContainer>
+        <Flex justifyContent='space-between'>
+          <StyledValue asTitle>{text}</StyledValue>
+
+          {!!value.details && <StyledValue asTitle>***{value.details.accountNumber}</StyledValue>}
```
Comment (julianromero-bc) [inline packages/blockchain-wallet-v4-frontend/src/modals/BuySell/LinkedPaymentAccounts/Accounts/Bank/index.tsx:56]: and details.. and all props extracted from value

**Review by julianromero-bc — CHANGES_REQUESTED** [review]

File: packages/blockchain-wallet-v4-frontend/src/modals/BuySell/LinkedPaymentAccounts/Accounts/Card/index.tsx [hunk 9]
```diff
@@ -23,27 +27,50 @@ type Props = {
...
+    <DisplayContainer
+      data-e2e={`sb${value.type.toLowerCase()}Cards`}
+      role='button'
+      onClick={onClick}
+    >
+      <DisplayIcon>{icon}</DisplayIcon>
+      <MultiRowContainer>
+        <Flex justifyContent='space-between'>
+          <StyledValue asTitle>{text.toLowerCase()}</StyledValue>
+          {!!value.card && <StyledTitle asValue>***{value.card?.number}</StyledTitle>}
```
Comment (julianromero-bc) [inline packages/blockchain-wallet-v4-frontend/src/modals/BuySell/LinkedPaymentAccounts/Accounts/Card/index.tsx:50]: be friend of destructuring it helps to reduce amount of code written

File: packages/blockchain-wallet-v4-frontend/src/components/Flyout/model.tsx [hunk 10]
```diff
@@ -603,29 +619,60 @@ const renderBank = (value: BSPaymentMethodType | BankTransferAccountType | Benef
...
+    if (label) {
+      return label.toLocaleLowerCase()
+    }
+  }
+
+  return 'Credit or Debit Card'
+}
+
+const renderCardSubtitle = (value: BSPaymentMethodType) => {
+  if (value.limits) {
```
Comment (julianromero-bc) [inline packages/blockchain-wallet-v4-frontend/src/components/Flyout/model.tsx:640]: missing destructuring

**Review by pedroapfilho — APPROVED** [review]

**Review by julianromero-bc — APPROVED** [review]

File: packages/blockchain-wallet-v4-frontend/src/scenes/Settings/General/LinkedBanks/template.success.tsx (hunk 1 above)
Comment (ASISBusiness) [reply packages/blockchain-wallet-v4-frontend/src/scenes/Settings/General/LinkedBanks/template.success.tsx]: > Would it make more sense to assign this to a variable in the function and not returning it? At a glance this looks like a ternary, not that there is a default value being returned if no `.limits` is present or no `paymentMethods.methods` matching the find are present.
> 
> Also, why are these the default limits? I mean, this must be specified somewhere but it does look like magic numbers, it being "hidden" here.
> 
> Doing something like `return (bankPaymentMethod?.limits || DEFAULT_LIMIT)` can be a bit more expressive..

**Pull request merged**